TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -391,5 +391,150 @@ public LockTableStatement CreateLockTableStatement(CodeElementsParser.LockTableS
             var savepointName = context.Diagnostics == null ? new SymbolReference(new AlphanumericValue((Token)context.savepoint_name), SymbolType.SqlIdentifier) : null;
             return new ReleaseSavepointStatement(savepointName);
         }
+        public enum AlterSequenceClauseTypes 
+        {
+            MinValue,
+            MaxValue,
+            Cycle,
+            Ordered,
+            Cache,
+            Restart,
+            Increment
+        }
+
+        public AlterSequenceStatement CreateAlterSequenceStatement(
```

Comment:
Ok this method is complex with the duplicate handling. Here is what I suggest

```csharp
        public AlterSequenceStatement CreateAlterSequenceStatement(CodeElementsParser.AlterSequenceStatementContext context)
        {
            var duplicates = new List<ParserRuleContextWithDiagnostics>();

            var clauseSet = new HashSet<AlterSequenceClauseType>();

            SyntaxProperty<bool> cycle = null;
            SqlConstant minValue = null;

            foreach (var alterSequenceClauseContext in context.alterSequenceClause())
            {
                if (alterSequenceClauseContext.minValueClause() != null)
                {
                    var minValueClause = alterSequenceClauseContext.minValueClause();
                    if (clauseSet.Add(AlterSequenceClauseType.MinValue))
                    {
                        System.Diagnostics.Debug.Assert(minValue == null);
                        minValue = new SqlConstant((Token)minValueClause.numeric_constant);
                    }
                    else
                    {
                        System.Diagnostics.Debug.Assert(minValue != null);
                        duplicates.Add(minValueClause);
                    }
                }
                else if (alterSequenceClauseContext.cycle() != null)
                {
                    var cycleClause = alterSequenceClauseContext.cycle();
                    if (clauseSet.Add(AlterSequenceClauseType.Cycle))
                    {
                        System.Diagnostics.Debug.Assert(cycle == null);
                        cycle = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(cycleClause));
                    }
                    else
                    {
                        System.Diagnostics.Debug.Assert(cycle != null);
                        duplicates.Add(cycleClause);
                    }
                }
                // else if (...)
                // {
                //     ...
                // }
                else if (alterSequenceClauseContext.noClauses() != null)
                {
                    if (alterSequenceClauseContext.noClauses().minvalue() != null)
                    {
                        var noMinValueClause = alterSequenceClauseContext.noClauses().minvalue();
                        if (clauseSet.Add(AlterSequenceClauseType.MinValue))
                        {
                            System.Diagnostics.Debug.Assert(minValue == null);
                        }
                        else
                        {
                            System.Diagnostics.Debug.Assert(minValue != null);
                            duplicates.Add(noMinValueClause);
                        }
                    }
                    else if (alterSequenceClauseContext.noClauses().cycle() != null)
                    {
                        var noCycleClause = alterSequenceClauseContext.noClauses().cycle();
                        if (clauseSet.Add(AlterSequenceClauseType.Cycle))
                        {
                            System.Diagnostics.Debug.Assert(cycle == null);
                            cycle = new SyntaxProperty<bool>(false, ParseTreeUtils.GetTokenFromTerminalNode(alterSequenceClauseContext.noClauses().SQL_NO()));
                        }
                        else
                        {
                            System.Diagnostics.Debug.Assert(cycle != null);
                            duplicates.Add(noCycleClause);
                        }
                    }
                    // else if (...)
                    // {
                    //    ...
                    // }
                }
            }
        }
```

Could you see how it looks with the whole set of options and try to look for factorizations  (use local function for examples)